homeassistant/components/camera/media_source.py

```diff
@@ -71,36 +73,42 @@ async def async_browse_media(
 
         can_stream_hls = "stream" in self.hass.config.components
 
-        # Root. List cameras.
-        component: EntityComponent[Camera] = self.hass.data[DOMAIN]
-        children = []
-        not_shown = 0
-        for camera in component.entities:
+        async def _media_source_for_camera(camera: Camera) -> BrowseMediaSource | None:
```

Comment:
I think everything looks good to go here, but now i'm thinking about one more idea to further streamline this, but not positive it would work out: Could the scope of this function be to just filter out the not relevant cameras and leave the browse media source building all together how it was?  It could just get the eligible cameras in one pass then build the sources in a simple loop together with building the root.

Reply:
We could return the camera instead of `BrowseMediaSource`, and then iterate over the results to build the sources, but I'm not sure I see why it would be better?

Reply:
Its possibly a style preference, but my thinking is: in general, reducing # of verical lines of inline functions or reducing the scope as much as possible seems positive. Having the `BrowseMediaSource` building closer together together seems logical as well. One exception may be if there is shared logic on both parts but i think the existing stream type filtering looks more complex than it needs to be

Reply:
It's a bit tricky because we also have to recalculate the content type (or return a tuple).
Arguably refactoring the creation of the `BrowseMediaSource` to a separate function would achieve a similar result (see my latest commit)

Reply:
Thanks, that looks good. What do you think about gutting all the existing vertical whitespace? It seems like it makes this harder to follow given it has so many branches and is in an inline function. Collapsing vertically will also help make this simpler to grok quickly.